source/extensions/tracers/datadog/dict_util.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <datadog/dict_reader.h>
+#include <datadog/dict_writer.h>
+
+#include <string>
+
+namespace Envoy {
+namespace Tracing {
+class TraceContext;
+} // namespace Tracing
+namespace Http {
+class RequestHeaderMap;
+class ResponseHeaderMap;
+} // namespace Http
+namespace Extensions {
+namespace Tracers {
+namespace Datadog {
+
+class RequestHeaderWriter : public datadog::tracing::DictWriter {
+  Http::RequestHeaderMap& headers_;
+
+public:
+  explicit RequestHeaderWriter(Http::RequestHeaderMap& headers);
+
+  void set(datadog::tracing::StringView key, datadog::tracing::StringView value) override;
+};
+
+class ResponseHeaderReader : public datadog::tracing::DictReader {
+  const Http::ResponseHeaderMap& headers_;
```

Comment:
style-guide nit: put the private members at the bottom, in a `private:` section -- what you have is totally fine it's just for consistency with the codebase and style guide.